src/components/FiscalYear/FiscalYearDetails/FiscalYearDetailsContainer.js

```diff
@@ -23,32 +21,42 @@ import FiscalYearDetails from './FiscalYearDetails';
 
 const FiscalYearDetailsContainer = ({
   mutator,
-  resources,
   match,
   onClose,
   history,
 }) => {
   const fiscalYearId = match.params.id;
+  const [fiscalYear, setFiscalYear] = useState({});
+  const [funds, setFunds] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const showToast = useShowToast();
 
   useEffect(
     () => {
-      mutator.fiscalYear.reset();
-      mutator.ledgers.reset();
-      mutator.funds.reset();
-      mutator.groups.reset();
-
-      mutator.fiscalYear.GET();
-      mutator.ledgers.GET();
-      mutator.funds.GET();
-      mutator.groups.GET();
+      setIsLoading(true);
+      mutator.fiscalYear.GET()
+        .then(fy => {
+          setFiscalYear(fy);
+
+          const fyFundsPromise = mutator.fyFunds.GET({
+            params: {
+              query: `budget.fiscalYearId=${fy.id}`,
+              limit: LIMIT_MAX,
+            },
+          });
+
+          return fy.id ? fyFundsPromise : [];
+        })
+        .then(response => setFunds(response))
+        .catch(() => {
+          showToast('ui-finance.fiscalYear.actions.load.error', 'error');
```

Comment:
it's better to reset state also, since old data will be displayed

Reply:
or maybe it's even better to do after `setIsLoading(true);`

Reply:
Thanx. Missed it.